src/main/java/com/github/invictum/dto/AbstractDto.java

```diff
@@ -14,27 +14,57 @@
 public class AbstractDto {
 
     private final static Logger LOG = LoggerFactory.getLogger(AbstractDto.class);
+    private String thisClassName = "AbstractDto";
 
     public static final boolean FULL_DTO_VIEW = Boolean
             .valueOf(PropertiesUtil.getProperty(EnhancedSystemProperty.FullDtoView));
 
     private String extractData(Object object, Field attribute) {
         attribute.setAccessible(true);
+
+        Object value;
         try {
-            return (String) attribute.get(object);
-        } catch (IllegalAccessException e) {
+            value = attribute.get(object);
+        } catch (IllegalAccessException var4) {
             LOG.error("Failed to get data for {} attribute", attribute);
+            return null;
+        }
+
+        String result = null;
+        if (value instanceof String) {
+            result = (String)value;
+        }
+        if (value instanceof Integer) {
+            result = Integer.toString((Integer)value);
         }
-        return null;
+        if (value instanceof Float) {
+            result = Float.toString((Float)value);
+        }
+        if (value instanceof Boolean) {
+            result = Boolean.toString((Boolean)value);
+        }
+
+        return result;
     }
 
     private List<Attribute> getData(Object object, boolean includeNulls) {
         List<Attribute> data = new ArrayList<>();
-        for (Field field : object.getClass().getDeclaredFields()) {
+        Field[] fields = new Field[0];
```

Comment:
I think here we can simplify code with recursive fields collection.
private Field[] getFields(Class class) {
        Field[] fields = class.getDeclaredFields();
        return class.getSuperclass() == AbstractDto.class ? fields : (Field[]) ArrayUtils
                .addAll(fields, getFields(class.getSuperclass()));
    }